src/Rule/Ip.php

```diff
@@ -34,14 +34,16 @@ final class Ip implements RuleWithOptionsInterface, SkipOnErrorInterface, WhenIn
     use WhenTrait;
 
     /**
-     * Negation char.
+     * Negation character.
      *
-     * Used to negate {@see $ranges} or {@see $network} or to negate validating value when {@see $allowNegation}
+     * Used to negate {@see $ranges} or {@see $network} or to negate value validated when {@see $allowNegation}
      * is used.
      */
-    private const NEGATION_CHAR = '!';
+    private const NEGATION_CHARACTER = '!';
```

Comment:
```suggestion
    public const NEGATION_CHARACTER = '!';
```

Reply:
Why making it public?

Reply:
I think we should allow duplication in the internal implementation in this case to lower the number of useless things available for end user.